### PR TITLE
[GSK-1953] Max data size as param

### PR DESCRIPTION
**Pull request by luca-rossi**
This PR allow users to set maximum number of samples (`max_dataset_size`) in loss-based scan detectors.

### CODE REVIEW

File: giskard/scanner/common/loss_based_detector.py [hunk 1]
```diff
@@ -23,6 +23,9 @@ class LossBasedDetector(Detector):
 
     _needs_target = True
 
+    def __init__(self, max_dataset_size: int = None):
```
Comment (mattbit) [inline giskard/scanner/common/loss_based_detector.py:26]: Set default directly

File: giskard/scanner/common/loss_based_detector.py [hunk 2]
```diff
@@ -39,7 +42,7 @@ def run(self, model: BaseModel, dataset: Dataset, **kwargs):
             return []
 
         # If the dataset is very large, limit to a subsample
-        max_data_size = self.MAX_DATASET_SIZE // len(model.meta.feature_names or dataset.columns)
+        max_data_size = self.max_dataset_size // len(model.meta.feature_names or dataset.columns)
```
Comment (mattbit) [inline giskard/scanner/common/loss_based_detector.py:45]: skip the division as discussed

File: giskard/scanner/common/loss_based_detector.py [hunk 3]
```diff
@@ -23,6 +23,9 @@ class LossBasedDetector(Detector):
 
     _needs_target = True
 
+    def __init__(self, max_dataset_size: int = MAX_DATASET_SIZE):
```
Comment (mattbit) [inline giskard/scanner/common/loss_based_detector.py:26]: inline this and probably divide by ~10?

**Review by mattbit — CHANGES_REQUESTED** [review]
All good, one small change

File: giskard/scanner/common/loss_based_detector.py (hunk 3 above)
Comment (luca-rossi) [reply giskard/scanner/common/loss_based_detector.py:26]: Ok, also maybe it still makes sense to divide by n_features if the param is not specified?

Comment (mattbit) [reply giskard/scanner/common/loss_based_detector.py:26]: yep, you could set `Optional[int] = None` and calculate the default at runtime if not provided

**Comment by mattbit** [comment]
Broken test!

**Review by mattbit — COMMENTED** [review]
Ok, but please fix conflicts

**Review by mattbit — APPROVED** [review]

**Pull request merged**